flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -1038,7 +1038,8 @@ private boolean transitionState(ExecutionState currentState, ExecutionState targ
 		}
 
 		if (STATE_UPDATER.compareAndSet(this, currentState, targetState)) {
-			markTimestamp(targetState);
+			long timestamp = System.currentTimeMillis();
+			markTimestamp(targetState, timestamp);
```

Comment:
Can we let `markTimestamp` return the timestamp and move `System.currentTimeMillis` back to it?